src/bin.rs

```diff
@@ -109,6 +109,13 @@ pub async fn run() -> CabotResult<()> {
                 .default_value("10")
                 .help("timeout for the tcp read in seconds"),
         )
+        .arg(
+            Arg::with_name("REQUEST_TIMEOUT")
+                .long("timeout")
+                .takes_value(true)
+                .default_value("30")
```

Comment:
There should not be a default value for a timeout for the entire request. Otherwise downloading large files that take more than 30 seconds to download will be impossible by default, and that behaviour is counter-intuitive.

Reply:
You are right (again :smiley: ). I've set it to `0` in master...